### PR TITLE
fix(runtime): rewrite with typescript 

### PR DESCRIPTION
**Pull request by Baoyuantop**

### CODE REVIEW

**Comment by Baoyuantop** [comment]
passed the sqlite test 🚀

**Comment by Baoyuantop** [comment]
TBR @tiannian

File: runtime/sqlite/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      const { ptr, length } = setValue("fail");
+      wasm_exports.call_sql_callback_fn(ptr, length, fn, addr);
+    } else {
+      const { ptr, length } = setValue("ok");
+      wasm_exports.call_sql_callback_fn(ptr, length, fn, addr);
+    }
+  })
+};
+
+export const _sql_query_callback = (ptr: number, path_length: number, fn: any, addr: any) => {
```
Comment (tiannian) [inline runtime/sqlite/index.ts]: `fn` and `addr` also is number. arguments are all number in wasm.

**Comment by tiannian** [comment]
I find a binary file in source code. @Baoyuantop

**Comment by Baoyuantop** [comment]
bros try

**Review by tiannian — APPROVED** [review]

**Comment by tiannian** [comment]
bors merge

**Comment by tiannian** [comment]
@Baoyuantop build failed due to npm.

**Comment by tiannian** [comment]
bors try

**Comment by Baoyuantop** [comment]
bors try

**Comment by Baoyuantop** [comment]
bors merge

**Pull request merged**